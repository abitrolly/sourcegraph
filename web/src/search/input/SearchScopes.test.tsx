import { createMemoryHistory } from 'history'
import React from 'react'
import { MemoryRouter } from 'react-router'
import { SearchScopes } from './SearchScopes'
import * as GQL from '../../../../shared/src/graphql/schema'
import { Settings } from '../../schema/settings.schema'
import { mount } from 'enzyme'

const BASE_PROPS = {
    authenticatedUser: {},
    history: createMemoryHistory(),
    query: 'abc',
    patternType: GQL.SearchPatternType.literal,
    versionContext: undefined,
}

describe('SearchScopes', () => {
    test('empty', () =>
        expect(
            mount(
                <MemoryRouter>
                    <SearchScopes {...BASE_PROPS} settingsCascade={{ final: {}, subjects: [] }} />
                </MemoryRouter>
            )
        ).toMatchSnapshot())

    test('with scopes', () => {
        const settings: Settings = { 'search.scopes': [{ name: 'n', value: 'v', description: 'd', id: 'i' }] }
        expect(
            mount(
                <MemoryRouter>
                    <SearchScopes {...BASE_PROPS} settingsCascade={{ final: settings, subjects: [] }} />
                </MemoryRouter>
            )
        ).toMatchSnapshot()
    })
})
