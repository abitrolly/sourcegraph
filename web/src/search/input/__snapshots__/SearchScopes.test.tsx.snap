// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScopes empty 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <SearchScopes
      authenticatedUser={Object {}}
      history="[History]"
      patternType="literal"
      query="abc"
      settingsCascade={
        Object {
          "final": Object {},
          "subjects": Array [],
        }
      }
    >
      <div
        className="search-scopes__edit"
      >
        <NavLink
          className="search-scopes__add-edit"
          to="/settings"
        >
          <Link
            aria-current={null}
            className="search-scopes__add-edit"
            to={
              Object {
                "hash": "",
                "key": null,
                "pathname": "/settings",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="search-scopes__add-edit"
              href="/settings"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="search-scopes__add-edit"
                href="/settings"
                onClick={[Function]}
              >
                <small
                  className="search-scopes__center"
                >
                  <span
                    className="search-scopes__add-scopes"
                  >
                    Add search scopes for quick filtering
                  </span>
                </small>
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
      </div>
    </SearchScopes>
  </Router>
</MemoryRouter>
`;

exports[`SearchScopes with scopes 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <SearchScopes
      authenticatedUser={Object {}}
      history="[History]"
      patternType="literal"
      query="abc"
      settingsCascade={
        Object {
          "final": Object {
            "search.scopes": Array [
              Object {
                "description": "d",
                "id": "i",
                "name": "n",
                "value": "v",
              },
            ],
          },
          "subjects": Array [],
        }
      }
    >
      <FilterChip
        key="0"
        name="n"
        onFilterChosen={[Function]}
        query="abc"
        value="v"
      >
        <button
          className="btn btn-sm text-nowrap filter-chip "
          data-testid="filter-chip"
          onClick={[Function]}
          onMouseDown={[Function]}
          title="v"
          type="button"
          value="v"
        >
          <div>
            n
          </div>
        </button>
      </FilterChip>
      <div
        className="search-scopes__edit"
      >
        <NavLink
          className="search-scopes__add-edit"
          to="/settings"
        >
          <Link
            aria-current={null}
            className="search-scopes__add-edit"
            to={
              Object {
                "hash": "",
                "key": null,
                "pathname": "/settings",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="search-scopes__add-edit"
              href="/settings"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="search-scopes__add-edit"
                href="/settings"
                onClick={[Function]}
              >
                <small
                  className="search-scopes__center"
                >
                  Edit
                </small>
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
      </div>
    </SearchScopes>
  </Router>
</MemoryRouter>
`;
